qiskit/primitives/base_estimator.py

```diff
@@ -296,7 +297,7 @@ def __call__(
                 if not isinstance(circuit, (int, np.integer))
                 else circuit
                 for circuit in circuits
-            ]
+            ]  # TODO: typed as list of int after this operations
```

Comment:
What does this comment mean? It's ambiguous to me. Either you postpone the type annotation of `circuits`, or you postpone the type casting to `List[int]`.

Reply:
Initially `circuits` is typed as `Sequence[int | QuantumCircuit]`, but after this line it is `List[int]`. This need to be known by `mypy` since afterward precense of `QuantumCircuit` in type leads to type errors

Reply:
OK I just saw that you wrote
> Large part of those errors can be fixed if the type cast is made explicit (I've added todo in corresponding places).

So, you meant it needs to be casted. But this context of what you want to do with this line, is not known to the people reading just the code.

Reply:
Yeah, I think you just need to clarify the comment in the way you just described in your reply.